src/hera/security_context.py

```diff
@@ -0,0 +1,78 @@
+from typing import List, Optional
+
+from argo_workflows.model.capabilities import Capabilities
+from argo_workflows.model.pod_security_context import PodSecurityContext
+from argo_workflows.model.security_context import SecurityContext
+from pydantic import BaseModel
+
+
+class WorkflowSecurityContext(BaseModel):
```

Comment:
I think there's some opportunity for restructuring here and I am curious what you think about the following suggestion:
* make a parent `SecurityContext` class that has the 4 permission fields (`run_as_user`, `run_as_group`, `fs_group`, `run_as_non_root`)
* make the `WorkflowSecurityContext` inherit from `SecurityContext` with no additional code added
* make the `TaskSecurityContext` inherit from `SecurityContext` with the extra `additional_capabilities` field

So, the code turns into:
```python
from pydantic import BaseModel
from argo_workflows.model import (
    Capabilities,
    PodSecurityContext,
    SecurityContext,
)

class SecurityContext(BaseModel):
    run_as_user: Optional[int] = None
    run_as_group: Optional[int] = None
    fs_group: Optional[int] = None
    run_as_non_root: Optional[bool] = None
    
    def get_context(...): 
        # current implementation

class WorkflowSecurityContext(SecurityContext):
    pass

class TaskSecurityContext(SecurityContext):
    additional_capabilities: List[str] = None

    def get_context(...): 
        # different implementation  
```

Reply:
As long as [this](https://github.com/kubernetes/kubernetes/blob/master/pkg/securitycontext/util.go#L163-L196) is true in k8s, we can do this abstraction. Though it might represent a false knowledge about the parent API, because the `SecurityContext` and `PodSecurityContext` in argo are not subclasses of any parent abstract class, what do you think?